### PR TITLE
feat(ui): reuse transaction when applying updates (#295)

### LINKED ISSUE #295: Reuse the transaction from the update dialog when applying updates via UI
When updating packages, the UI currently creates a transaction for showing the outdated packages and another transaction when applying the updates. 

In an edge case where a package is updated between the time the dialog is opened and the time the confirm button is pressed, another version than the one the user has agreed to will be installed. 

This could be mitigated by using the same transaction for both but it is not so straight forward, since multiple connected clients doing the same thing should be (however unlikely, given our use-case) considered and handled correctly.

### PR DESCRIPTION
**Pull request by christophenne**
Closes #295

## ✅ Checks
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required

### CODE REVIEW

**Review by kosmoz — CHANGES_REQUESTED** [review]
Here are some comments about the implementation.

More generally, I feel like this is not really reusing the existing transaction but validating that nothing has changed in the time between showing the dialog and confirming the update. Related to that, I would have expected that the old transaction to still be applied in case something has changed in the meantime (of course the "update available" banner would still be visible).

With that said, I still think this is a good thing to have but I would keep the issue open for now.

File: internal/web/templates/components/pkg-update-modal.html [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {{ define "pkg-update-modal" }}
   <div class="modal-dialog modal-dialog-centered" id="pkg-update-modal">
     <div class="modal-content">
-      <form hx-post="/packages/update">
+      <form hx-post="/packages/update?{{ .UpdateTransactionParams }}">
```
[the remark this answers is not in the document]
Comment (christophenne) [reply internal/web/templates/components/pkg-update-modal.html:4]: I dislike both, and both seem non-idiomatic to me :smiley: I can change it though.

[the remark this answers is not in the document]
Comment (kosmoz) [reply internal/web/templates/components/pkg-update-modal.html:4]: I think this is exactly what `<input type="hidden">` is for though :thinking:

**Comment by kosmoz** [comment]
> True, it's a validation. It's pragmatic but I doubted that reusing the old transaction is worth the implementation effort of either holding something server-side or changing the updater to support version-specific updates for multiple packages. That's the two approaches I saw, but I decided against them because I thought this validation is probably good enough – it's not like this is going to happen too often. I don't know to be honest, it's all not that great.
> 
> But if we decide that we want to do the transaction-reuse thing, I would do it right away instead of this validation.

Couldn't we just store a `map[int]update.Transaction` or something like that in the `server` and let the update modal form tell us the key of the transaction it should apply after confirmation (using an `<input type="hidden">` :smile:)? For the key we can use `rand.Int()`.

**Comment by christophenne** [comment]
> > True, it's a validation. It's pragmatic but I doubted that reusing the old transaction is worth the implementation effort of either holding something server-side or changing the updater to support version-specific updates for multiple packages. That's the two approaches I saw, but I decided against them because I thought this validation is probably good enough – it's not like this is going to happen too often. I don't know to be honest, it's all not that great.
> > But if we decide that we want to do the transaction-reuse thing, I would do it right away instead of this validation.
> 
> Couldn't we just store a `map[int]update.Transaction` or something like that in the `server` and let the update modal form tell us the key of the transaction it should apply after confirmation (using an `<input type="hidden">` 😄)? For the key we can use `rand.Int()`.

Fair enough, changed it.

**Review by kosmoz — APPROVED** [review]
Looks good!

**Pull request merged**